### PR TITLE
[relay-runtime] Move PreloadableConcreteRequest from react-relay to relay-runtime

### PR DESCRIPTION
**Pull request by tobias-tengler**
I'm working on adding `@preloadable` support to the OSS version of Relay: https://github.com/facebook/relay/pull/4515
The generated `$parameters.ts` file needs to import the `PreloadableConcreteRequest` type. Currently this type is defined in `@types/react-relay` for seemingly no reason. The other request types are defined in `@types/relay-runtime`.
This PR moves `PreloadableConcreteRequest` to `@types/relay-runtime` so the `relay-compiler` doesn't have to suddenly start importing types from `react-relay`. 

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/relay/pull/4110#issuecomment-1771372672

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-relay ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-relay@latest/))
was missing the following properties:
1. MutationTypes
1. RangeOperations
## relay-runtime ([<kbd>unpkg</kbd>](https://unpkg.com/browse/relay-runtime@latest/))
was missing the following properties:
1. MutationTypes
1. RangeOperations
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c8dfa0b1858a0fc43870f62b6cea7627b869ed22
</p>

**Review by maraisr — COMMENTED** [review]
Nice work mate, thank you so much! Left one comment

File: types/relay-runtime/lib/util/RelayConcreteNode.d.ts
Comment (tobias-tengler) [reply types/relay-runtime/lib/util/RelayConcreteNode.d.ts:21]: Why, because of an older TypeScript version? The `kind` should only allow that specific string literal and it has already (correctly) been that way in the `react-relay` package.

Comment (maraisr) [reply types/relay-runtime/lib/util/RelayConcreteNode.d.ts:21]: Yeah I know, but I'm not sure of the consequences. OSS has never had this supported, so we don't know what will break. All the other request types has the kind set to string for reasons i cannot recall right now.

Comment (tobias-tengler) [reply types/relay-runtime/lib/util/RelayConcreteNode.d.ts]: Got it! Aligned it to the other request types in 28d0f0c1a56fc69a37c652a5224a3aac9a9eee81

**Review by maraisr — APPROVED** [review]
Looks good to me ✨ thank you so much

**Comment by maraisr** [comment]
Think we might wait for this to land on the upstream, before merging this.

**Pull request closed without merging**
